client/signup/steps/p2-complete-profile/index.jsx

```
@@ -0,0 +1,260 @@
+				const errorMessage = props.step.errors[ 0 ].message;
+
+				this.logValidationErrorToLogstash( ERROR_CODE_FROM_LOCAL_STORAGE, errorMessage );
+
+				initialState = formState.setFieldErrors(
+					formState.setFieldsValidating( initialState ),
+					{
+						site: {
+							[ ERROR_CODE_FROM_LOCAL_STORAGE ]: errorMessage,
+						},
+					},
+					true
+				);
+			}
+		} else {
+			initialState = {
+				fullName: {
+					value: '',
+				},
+			};
+		}
+
+		this.formStateController = new formState.Controller( {
+			fieldNames: [ 'fullName' ],
+			sanitizerFunction: this.sanitize,
+			validatorFunction: this.validate,
+			onNewState: ( state ) => {
+				this.setState( { form: state } );
+			},
+			onError: this.handleFormControllerError,
+			debounceWait: VALIDATION_DELAY_AFTER_FIELD_CHANGES,
+			hideFieldErrorsOnChange: true,
+			initialState: initialState,
+			skipSanitizeAndValidateOnFieldChange: true,
+		} );
+
+		this.state = {
+			form: this.formStateController.getInitialState(),
+			isSubmitting: false,
+			suggestedSubdomains: [],
+			lastInvalidSite: '',
+		};
+	}
+
+	componentWillUnmount() {
+		this.save();
+	}
+
+	validate = ( fields, onComplete ) => {
+		const messages = {};
+
+		if ( isEmpty( fields.fullName ) || fields.fullName.length < 3 ) {
+			messages.fullName = {
+				[ ERROR_CODE_MISSING_FULL_NAME ]: this.props.translate(
+					'Please enter your full name (3 characters or more).'
+				),
+			};
+		}
+
+		onComplete( null, messages );
+	};
+
+	handleSubmit = ( event ) => {
+		event.preventDefault();
+
+		this.setState( { isSubmitting: true } );
+
+		this.formStateController.handleSubmit( ( hasErrors ) => {
+			const fullName = formState.getFieldValue( this.state.form, 'fullName' );
+
+			if ( hasErrors ) {
+				this.setState( { isSubmitting: false } );
+
+				return;
+			}
+
+			recordTracksEvent( 'calypso_signup_p2_complete_profile_step_submit' );
+
+			// API calls
+
+			this.props.saveUserSettings( { display_name: fullName } );
```

What happens if this request fails? Are we handling it in any way?

---

Right now, the Calypso's data layer handles it by showing a "There was a problem saving your changes." notice. Not sure how rare it would be and whether we need to intercept it in this step and tell user to try again or something or just continue the flow as we do now.

---

IMHO, the probability for it to fail is the same as any other request. The difference is that we do handle some of those failures gracefully, while here, we'd likely fail silently (silently for the user at least).

---

Yep, probability is the same but I guess what I meant is that this is not an important request. Worst case user will not have their name set. The whole step is kind of additional. But yeah, I'll try to intercept the error and stop the submit in that case.